posthog/api/plugin.py

```diff
@@ -116,11 +116,31 @@ def _update_plugin_attachment(plugin_config: PluginConfig, key: str, file: Optio
             plugin_attachment.file_size = file.size
             plugin_attachment.contents = file.file.read()
             plugin_attachment.save()
+
+            log_activity(
+                organization_id=plugin_config.team.organization.id,
+                team_id=plugin_config.team.id,
+                user=user,
+                item_id=plugin_attachment.id,
+                scope="PluginAttachment",
+                activity="attachment_updated",
+                detail=Detail(name=file.name),
+            )
         else:
             plugin_attachment.delete()
+
+            log_activity(
+                organization_id=plugin_config.team.organization.id,
+                team_id=plugin_config.team.id,
+                user=user,
+                item_id=plugin_attachment.id,
+                scope="PluginAttachment",
```

Comment:
For this to be more useful we'd like this to be under the PluginConfig scope - these are concepts that are a bit annoying to wrap your head around at first, but, to use an analogy, the `Plugin` model is like a class (i.e. the plugin definition) and the `PluginConfig` (despite the slightly misleading name) is an instance of the `Plugin` class.

Any time a team enables a plugin for themselves, we create a new PluginConfig, and that's what maps 1:1 to plugin VMs. 

Plugin configs in turn can be configured with some JSON fields and attachments. 

All this to say that "attachments" are a level lower than we'd like to log activity for - in reality we want to know the changes to attachments from a PluginConfig level. i.e. "Attachment X changed for plugin config Y".

Hope this makes sense. I do realize how it's very confusing though, particularly as we seem to map activities to models. In this case attachments are just a part of plugin config in practice though.